sn_networking/src/cmd.rs

```diff
@@ -158,7 +158,26 @@ impl SwarmDriver {
                 let _ = sender.send(peers);
             }
             SwarmCmd::AddKeysToReplicationFetcher { peer, keys, sender } => {
-                let keys_to_fetch = self.replication_fetcher.add_keys(peer, keys);
+                // check if we have any of the data before adding it.
+                let existing_keys: HashSet<NetworkAddress> = self
```

Comment:
node level already carried out the filtering.

This block of filtering code was originally in SwarmDriver.
But got moved up to Node when moving the whole replication flow up to Node.

It maybe good to have it here, so the knowledge is more synced.
But in that case, the filtering in Node shall then be removed to avoid un-necessary work.